hermes/probe/read/read.go

```diff
@@ -0,0 +1,124 @@
+	if fileID < minFileID || fileID > maxFileID {
+		return fmt.Errorf("invalid argument: fileID = %d; want %d <= fileID <= %d", fileID, minFileID, maxFileID)
+	}
+	bucketName := target.Target.GetBucketName()
+
+	// Verify that the file is present in the State Journal
+	fileName, ok := target.Journal.Filenames[fileID]
+	if !ok {
+		return fmt.Errorf("file with the ID: %d is missing from the State Journal", fileID)
+	}
+
+	// Verify that the file that we want to read is in fact present in the target system
+	fileNamePrefix := fmt.Sprintf(FileNameFormat, fileID, "")
+	query := &storage.Query{Prefix: fileNamePrefix}
+	start := time.Now()
+	objIter := client.Bucket(bucketName).Objects(ctx, query)
+	var namesFound []string
+	for {
+		obj, err := objIter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("ReadFile(ID: %d) check failed due to: %w", fileID, err)
+		}
+		namesFound = append(namesFound, obj.Name)
+	}
+	finish := time.Now()
+	if len(namesFound) == 0 {
+		target.LatencyMetrics.APICallLatency[metrics.APIListFiles][metrics.FileMissing].Metric(hermesAPILatencySeconds).AddFloat64(finish.Sub(start).Seconds())
+		return fmt.Errorf("ReadFile(ID: %d) could not read file as the file with the provided ID does not exist on the target system", fileID)
+	}
+	if len(namesFound) != 1 {
+		return fmt.Errorf("expected exactly one file in bucket %q with prefix %q; found %d: %v", bucketName, fileNamePrefix, len(namesFound), namesFound)
+	}
+	if namesFound[0] != fileName {
+		return fmt.Errorf("ReadFile(ID %d) check failed expected file name present %q got %q", fileID, fileName, namesFound[0])
+	}
+	target.LatencyMetrics.APICallLatency[metrics.APIListFiles][metrics.Success].Metric(hermesAPILatencySeconds).AddFloat64(finish.Sub(start).Seconds())
+
+	start = time.Now()
+	reader, err := client.Bucket(bucketName).Object(fileName).NewReader(ctx)
+	if err != nil {
+		var status metrics.ExitStatus
+		switch err {
+		case storage.ErrObjectNotExist:
+			status = metrics.FileMissing
+		case storage.ErrBucketNotExist:
+			status = metrics.BucketMissing
+		default:
+			status = metrics.ProbeFailed
+		}
+		target.LatencyMetrics.APICallLatency[metrics.APIGetFile][status].Metric("hermes_api_latency_s").AddFloat64(time.Now().Sub(start).Seconds())
+		return fmt.Errorf("readFile(id: %d).%v: could not read file %s: %w", fileID, status, fileName, err)
+	}
+	defer reader.Close()
+	h := sha1.New()
+	if _, err := io.Copy(h, reader); err != nil {
+		return fmt.Errorf("io.Copy: %v", err)
+	}
+	gotChecksum := fmt.Sprintf("%x", h.Sum(nil))
+	wantChecksum := fileName[len(fileNamePrefix):len(fileName)]
+	if gotChecksum != wantChecksum {
+		return fmt.Errorf("the calculated checksum: %q does not match the checksum in the file name: %q", gotChecksum, wantChecksum)
+	}
+	logger.Infof("verified consiistency for object %q in bucket %q.", fileName, bucketName)
```

Comment:
Typo.
Also, there's no need for the period at the end.

Reply:
fixed